test/framework/src/main/java/org/elasticsearch/test/rest/Stash.java

```diff
@@ -67,12 +70,12 @@ public void clear() {
      * The stash contains fields eventually extracted from previous responses that can be reused
      * as arguments for following requests (e.g. scroll_id)
      */
-    public boolean isStashedValue(Object key) {
-        if (key == null) {
-            return false;
-        }
+    public boolean containsStashedValue(Object key) {
+        if (key == null) return false;
         String stashKey = key.toString();
-        return Strings.hasLength(stashKey) && stashKey.startsWith("$");
+        if (false == Strings.hasLength(stashKey)) return false;
+        if (stashKey.startsWith("$")) return true;
+        return EXTENDED_KEY.matcher(stashKey).find();
```

Comment:
`${stashedKey}` always returns a string though. We totally use the stash for non-string things.